### PR TITLE
Cura 7900 python 39

### PR DESCRIPTION
**Pull request by jellespijker**
@evtrados, @rburema and @Ghostkeeper  I have created a PR to gather my comments

### CODE REVIEW

File: projects/Uranium.cmake
Comment (jellespijker) [inline projects/Uranium.cmake:7]: Shouldn't this be:
`find_package(PyQt 5.15.2 REQUIRED)`

**Review by jellespijker — COMMENTED** [review]
One tiny little change requested.
find_package(PyQt **5.15.2** ...

**Comment by Ghostkeeper** [comment]
We'll be entering a feature freeze tomorrow afaik. We will also inform the plug-in maintainers of all the changes then shortly afterwards, probably just before or around the beta release. After the beta there will be just over 2 weeks for the 4.9 release as normal, if everything goes according to plan.

**Comment by thopiekar** [comment]
Alright. Many thanks for the update @Ghostkeeper 👍

File: projects/Uranium.cmake
Comment (Ghostkeeper) [reply projects/Uranium.cmake:7]: Yes, that would indeed be more correct! I reckon this accidentally slipped in.

Even though it says it needs to find PyQt 5.14, it would still find the 5.15 version (since newer versions are allowed to be found if it doesn't specify `EXACT`).

File: projects/libCharon.cmake
Comment (Ghostkeeper) [inline projects/libCharon.cmake:6]: I see that you added this variable to CMake without any review here: https://github.com/Ultimaker/libCharon/commit/21c4161a07c6ef76d145b198ac5ba4989914355f
That should probably have been reviewed. It's not even in our control to maintain that repository (final say is with S-line firmware).

However it is not correct to add a variable there specific to Cura. That library is not specific to Cura and other projects wouldn't want to deal with our mess.

If you want the CMake script there to find a specific version of CPython, you can find it here and provide it to their CMake script through `-DPython3_EXECUTABLE=${Python3_EXECUTABLE}` or something of the sort.

File: scripts/python3.8/linux/build.sh
Comment (Ghostkeeper) [inline scripts/python3.8/linux/build.sh:17]: Should say `cura-build-environment:master`, right?

File: scripts/python3.8/linux/build.sh
Comment (Ghostkeeper) [inline scripts/python3.8/linux/build.sh:29]: This is incorrect to be having in our source code as default. Specify that in the build server all you like while testing, but don't put it as default in the source code. It won't be interesting at all any more in the future.

File: scripts/python3.8/linux/build.sh
Comment (Ghostkeeper) [inline scripts/python3.8/linux/build.sh:26]: By convention we normally take version 0.0.0 as default if not specified otherwise. Version 9.9.9 is a bit weird, especially if Cura 9 is eventually released.

File: scripts/python3.8/linux/build.sh
Comment (Ghostkeeper) [inline scripts/python3.8/linux/build.sh:41]: As far as I know this never needs to be off any more.

File: scripts/python3.8/linux/build.sh
Comment (Ghostkeeper) [inline scripts/python3.8/linux/build.sh:101]: As far as I know this never needs to be off any more.

File: scripts/python3.8/linux/build.sh
Comment (Ghostkeeper) [inline scripts/python3.8/linux/build.sh:1]: Pretty much all files in this repository are scripts. Adding a "scripts" folder really makes no sense in this regard. Is it not better to put this in the "docker" folder since that's what it's used for and specific to?

File: scripts/python3.8/linux/build_in_docker.sh
Comment (Ghostkeeper) [inline scripts/python3.8/linux/build_in_docker.sh:1]: (Same comment as above: Put in "docker" folder? The comment below would be self-evident then.)

File: scripts/python3.8/linux/build_in_docker.sh
Comment (Ghostkeeper) [inline scripts/python3.8/linux/build_in_docker.sh:47]: As far as I know this variable is always ON now and doesn't need to be a parameter.

File: signing/osx.cmake
Comment (Ghostkeeper) [inline signing/osx.cmake:11]: This file doesn't exist. It refers to outside of this repository?!?

**Review by Ghostkeeper — CHANGES_REQUESTED** [review]

File: projects/libCharon.cmake
Comment (evtrados) [reply projects/libCharon.cmake:6]: I had already asked Bryan Foley and he told me to create a PR for their review, which was accepted last Friday.

File: scripts/python3.8/linux/build.sh
Comment (evtrados) [reply scripts/python3.8/linux/build.sh:26]: Yes, I agree. Should be 0.0.0

File: scripts/python3.8/linux/build.sh
Comment (evtrados) [reply scripts/python3.8/linux/build.sh:29]: Yeah, indeed this slipped through. Should be 0.0.0 and an empty extra version string.

File: scripts/python3.8/linux/build.sh
Comment (evtrados) [reply scripts/python3.8/linux/build.sh:1]: I also agree on that. We do have a docker/ folder in cura-build-environment, so it's much better to have one in cura-build as well.

File: scripts/python3.8/linux/build_in_docker.sh
Comment (evtrados) [reply scripts/python3.8/linux/build_in_docker.sh:1]: Yes, I also agree.

File: scripts/python3.8/linux/build.sh
Comment (evtrados) [reply scripts/python3.8/linux/build.sh:41]: Sure, I am also fine with that.

File: scripts/python3.8/linux/build.sh
Comment (evtrados) [reply scripts/python3.8/linux/build.sh:101]: Sure, I am also fine with that.

File: scripts/python3.8/linux/build_in_docker.sh
Comment (evtrados) [reply scripts/python3.8/linux/build_in_docker.sh:47]: Sure, I am also fine with that.

File: signing/osx.cmake
Comment (evtrados) [reply signing/osx.cmake:11]: Yes, CMake would not allow me to create specific commands with regards to signing the .so and .dylib files, so the best solution I could find without bypassing the whole signing process of CMake was to have a file on the build machine that takes care of this.

File: scripts/python3.8/linux/build.sh
Comment (evtrados) [reply scripts/python3.8/linux/build.sh:17]: We can put it as master there, yes. We need to create one environment with that tag though :D

File: signing/osx.cmake
Comment (Ghostkeeper) [reply signing/osx.cmake:11]: This violates our license and defeats the purpose of having an open source build system. We are legally not allowed to merge this.

Instead, I'd suggest to place this .sh script inside of the repository and call it from CMake in the exact same way.

Comment (evtrados) [reply signing/osx.cmake]: OK, I decided to add the file (changed the 'osx' suffix to 'macOS' as this is the new official name after the launch of Big Sur) in the signing/ dir together with the cura.entitlements file which it references. I was a bit hesitant in the beginning to add the latter since it lists the entitlements we give to our app, but as far as I checked everyone can more or less create one such file using Xcode.

File: projects/libCharon.cmake
Comment (evtrados) [reply projects/libCharon.cmake]: I managed to remove the python version on the builder VM. It was probably installed by Rick while initiating the instance and installing packages via brew. I removed the dependencies these packages had from that python version and now this is not an issue anymore. LibCharon can now identify 'our' version of python and everything is built properly. I'll also talk to the firmware team so that they revert my previously accepted PR.

File: signing/osx.cmake
Comment (evtrados) [reply signing/osx.cmake]: > This violates our license. We are legally not allowed to merge this, as far as I can tell. It also defeats the purpose of having an open source build system since nobody could reproduce the build without having that script.
> 
> Instead, I'd suggest to place this .sh script inside of the repository and call it from CMake in the exact same way.

I totally agree. Even though no one would really be able to create a build with `-DSIGN_PACKAGE=ON `anyway, since they would not have the codesign identity that is required for the signing stage ;)

**Review by nallath — APPROVED** [review]

**Pull request merged**
